src/psygnal/_signal.py

```diff
@@ -1280,49 +1280,50 @@ def __call__(self, args: tuple[object, ...]) -> bool:
     def __eq__(self, other: object) -> bool:
         return (
             isinstance(other, _BoundMethodCaller)
-            and self._ref == other._ref
-            and self._method_name == other._method_name
+            and self._method_ref() == other._method_ref()
+            and self._obj_ref == other._obj_ref
```

Comment:
if `self._method_ref() == other._method_ref()` is evaluated to fail then `            and self._obj_ref == other._obj_ref` will not change anything.

Reply:
nice move :)